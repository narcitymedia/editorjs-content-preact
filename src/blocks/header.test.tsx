import React from 'react';
import * as renderer from 'react-test-renderer';
import { BlockType } from '..';

import Header from './header';

describe('Header', () => {
  it('should render the 0 level heading as h1', () => {
    const args = {
      text: "Heading 1",
      level: 0,
    }
    const tree = renderer
      .create(<Header type={BlockType.Delimiter} data={args} contentUniqueId="XXX-XXX" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render the 1 level heading as h1', () => {
    const args = {
      text: "Heading 1",
      level: 1,
    }
    const tree = renderer
      .create(<Header type={BlockType.Delimiter} data={args} contentUniqueId="XXX-XXX" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render the 2 level heading as h2', () => {
    const args = {
      text: "Heading 2",
      level: 2,
    }
    const tree = renderer
      .create(<Header type={BlockType.Delimiter} data={args} contentUniqueId="XXX-XXX" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
