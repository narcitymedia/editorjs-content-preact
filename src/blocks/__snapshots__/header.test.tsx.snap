// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delimeter should render the 0 level heading as h1 1`] = `
<h1
  dangerouslySetInnerHTML={
    Object {
      "__html": "Heading 1",
    }
  }
  id="XXX-XXX"
/>
`;

exports[`Delimeter should render the 1 level heading as h1 1`] = `
<h1
  dangerouslySetInnerHTML={
    Object {
      "__html": "Heading 1",
    }
  }
  id="XXX-XXX"
/>
`;

exports[`Delimeter should render the 2 level heading as h2 1`] = `
<h1
  dangerouslySetInnerHTML={
    Object {
      "__html": "Heading 2",
    }
  }
  id="XXX-XXX"
/>
`;
